### ISSUE TITLE
INTERNAL ERROR: Black produced code that is not equivalent to the source.

### ISSUE THREAD

**Issue by chebee7i** [psf/black#1663]
**Describe the bug** 

Running black on a file produces: `INTERNAL ERROR: Black produced code that is not equivalent to the source.`

**To Reproduce**

```console
$ cat demo.py
def __getattr__(self, attr):
    """Something.
    """
    raise AttributeError, ('%s %s' % (self.__class__.__name__, attr))

$ black -t py27 --diff demo.py
error: cannot format demo.py: INTERNAL ERROR: Black produced code that is not equivalent to the source.  Please report a bug on https://github.com/psf/black/issues.  This diff might be helpful: /tmp/blk__7di9438.log
Oh no! 💥 💔 💥
1 file would fail to reformat.

$ black --version
black, version 20.8b1
```

**Environment (please complete the following information):**

- Version: 20.8b1
- OS and Python version: [e.g. Linux/Python 3.8.5]

**Does this bug also happen on master?** 

Yes on master with a new virtualenv, but no when using the online formatter.

**Additional context**

Exact text of demo.py is:
```python
>>> open('demo.py').read()
'def __getattr__(self, attr):\n    """Something.\n    """\n    raise AttributeError, (\'%s %s\' % (self.__class__.__name__, attr))\n'
```

**Comment by chebee7i** [comment]
The code style is quite "old"...modernizing it was enough to get around the issue.

**Comment by esmail** [comment]
Did you check the diff mentioned at the end of the error message? Seems likely to be a duplicate of #1635.

**Comment by chebee7i** [comment]
The effect is the same, but that bug doesn't seem to manifest as an `INTERNAL ERROR`.  Do you want to close this as a duplicate?  I'm fine either way. :-)

**Comment by chebee7i** [comment]
For reference sake:
```console
$ cat /tmp/blk__7di9438.log
Oh no! 💥 💔 💥
--- src
+++ dst
@@ -28,11 +28,11 @@
       body=
         Expr(
           value=
             Constant(
               value=
-                b'Something.\n    ',  # bytes
+                b'Something.',  # bytes
             )  # /Constant
         )  # /Expr
         Raise(
           inst=
             BinOp(
```

**Comment by JelleZijlstra** [comment]
This can definitely be kept separate; this is a specific crash (which we should fix!) and #1635 is someone disagreeing with Black's style choices (which is harder to fix).

**Comment by JelleZijlstra** [comment]
This is fixed on the latest release
```
% black -t py27 --diff issue1663.py 
--- issue1663.py	2021-05-29 03:27:32.518534 +0000
+++ issue1663.py	2021-05-29 03:27:41.808162 +0000
@@ -1,5 +1,3 @@
 def __getattr__(self, attr):
-    """Something.
-    """
-    raise AttributeError, ('%s %s' % (self.__class__.__name__, attr))
-
+    """Something."""
+    raise AttributeError, ("%s %s" % (self.__class__.__name__, attr))
would reformat issue1663.py
All done! ✨ 🍰 ✨
1 file would be reformatted.
```
